sda/processor.go

```diff
@@ -124,10 +124,17 @@ func NewServiceProcessor(c *Context) *ServiceProcessor {
 	}
 }
 
-// RegisterMessage puts a new message in the message-handler
-// XXX More comments are needed as it's not clear whether RegisterMessage waits
-// for message for/from Clients or for/from Services.
-func (p *ServiceProcessor) RegisterMessage(f interface{}) error {
+// RegisterHandler will store the given handler that will be used by the service.
+// f must be a function that takes two input:
+//  * network.ServerIdentity: from whom the message is coming from
+//  * Pointer to a struct: message that the service is ready to handle.
+// f must have two return values:
+//  * Pointer to a struct: message that the service has generated as a reply and
+//  that will be sent to the requester (the sender).
+//  * Error in any case there is an error.
+// f can be used to treat internal service messages as well as external requests
```

Comment:
Reading this comment and rereading the code below: why don't we make f a function instead of an `interface{}` then? The comment wouldn't be necessary and it would be more clear how to use it. I'll try later if it works.